elijah/app.ts

```diff
@@ -0,0 +1,137 @@
+import fs from "fs";
+import readline from "node:readline/promises";
+import util from "util";
+
+const readFile = util.promisify(fs.readFile);
+
+type User = {
+  username: string;
+  displayName: string;
+  state: string;
+  friends: string[];
+};
+
+type Post = {
+  postId: string;
+  userId: string;
+  visibility: "public" | "friend";
+};
+
+let posts: Post[] = [];
+let users: User[] = [];
+
+const getPostById = (id: string) =>
+  posts.filter((post) => post.postId === id)[0];
+const getUserByUsername = (username: string) =>
+  users.filter((user) => user.username === username)[0];
+
+const getIsPostVisible = (postId: string, username: string) => {
+  const post = getPostById(postId);
+  const author = getUserByUsername(post.userId);
+  const user = getUserByUsername(username);
+
+  let hasAccess = post.visibility === "public";
+  hasAccess = hasAccess || author.username === user.username;
+  hasAccess = hasAccess || author.friends.includes(user.username);
+
+  return hasAccess;
+};
+
+const entry = async () => {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  let shouldContinue = true;
+  while (shouldContinue) {
+    const choice = await rl.question(`
+1. Load input data.
+2. Check visibility.
+3. Retrieve posts.
+4. Search users by location.
+5. Exit.
+Choice: `);
+
+    switch (choice) {
+      case "1": {
+        let postInfoPath = await rl.question(
+          "Path to post-info.txt (data/post-info.txt): "
+        );
+        let userInfoPath = await rl.question(
```

Comment:
Interesting way to loop the UI, I enjoy how clean typescript makes your code read.